android/jni/com/mapswithme/platform/Platform.hpp

```diff
@@ -49,12 +50,17 @@ class Platform : public ::Platform
 
   void SetGuiThread(std::unique_ptr<base::TaskLoop> guiThread);
 
+  void SaveToSecureStorage(std::string const & key, std::string const & value);
+  bool LoadFromSecureStorage(std::string const & key, std::string & value);
+  void RemoveFromSecureStorage(std::string const & key);
```

Comment:
Could you please extract all three methods to a class named SecureStorage, with methos Save, Load and Remove? And android Platform will store the instance of the class.